### DISCUSSION TITLE
I'm looking for help with the "external-authentication" option in a connection string

### DISCUSSION THREAD

**Discussion by eugenepik** [trinodb/trino-python-client#459] [Q&A] [answered]
I am successfully connecting to Trino servers using the Java CLI and following prompt:

`java -jar trino.jar --user=myuser --password --server=https://trino.myserver.org --external-authentication --catalog "mycatalog" --schema "myschema"
`
It asks for a password, then opens browser for authentication and successfully creates a command line interface to Trino.

Please help me configure the Python Trino connection that uses **external-authentication** option. Thank you.

**Comment by hashhar** [comment, marked as answer]
See the examples at https://github.com/trinodb/trino-python-client?tab=readme-ov-file#oauth2-authentication.

`external-authentication` is a JDBC driver property. In Python client we have different auth classes (you can also add your own) for each auth mechanism so it's just a matter of using the correct auth class in the `connect` call.

**Comment by eugenepik** [comment]
Excellent! The browser authentication worked well with the following code:

```
from trino.dbapi import connect
from trino.auth import OAuth2Authentication

conn = connect(
    host="trino.abc.org",
    port=443,  
    user="myuser",
    auth=OAuth2Authentication(),
    http_scheme="https",
    catalog="mycatalog",
    schema="myschema"
)

# Test the connection
cursor = conn.cursor()
query = "SELECT * FROM mytable LIMIT 1"  
try:
    cursor.execute(query)
    for row in cursor.fetchall():
        print(row)
    print("Query executed successfully.")
except Exception as e:
    print(f"Failed to execute query: {e}")
```

Also, note there is no semicolon at the end of the SQL string, otherwise it would give me an error `mismatched input ';'. Expecting: <EOF>`

Thanks!
